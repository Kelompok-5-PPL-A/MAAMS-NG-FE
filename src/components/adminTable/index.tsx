import React from 'react'
import { Item } from '../../components/types/adminTable'
import { useRouter } from 'next/router'

interface Props {
  data: Item[]
}

const AdminTable: React.FC<Props> = ({ data }) => {
  const router = useRouter()

  const handleRowClick = (id: string) => {
    router.push(`/validator/${id}`)
  }

  return (
    <div className='relative overflow-x-auto shadow-md m-12'>
      <table className='w-full text-sm text-left text-black border-collapse'>
        <thead className='text-xs uppercase bg-[#FBC707] text-black'>
          <tr>
            <th scope='col' className='px-6 py-3 w-1/5'>
              Judul
            </th>
            <th scope='col' className='px-6 py-3 w-1/5'>
              Pengguna
            </th>
            <th scope='col' className='px-6 py-3 w-1/5'>
              Topik
            </th>
            <th scope='col' className='px-6 py-3 w-1/5'>
              Waktu
            </th>
            <th scope='col' className='px-6 py-3 w-1/5'>
              <span className='sr-only'>Lihat</span>
            </th>
          </tr>
        </thead>
        <tbody>
          {data.map(({ id, title, displayed_title, user, tags, timestamp }) => (
            <tr
              key={id}
              className='bg-white hover:bg-gray-200 border border-[#FBC707] text-xs h-20 cursor-pointer'
              onClick={() => handleRowClick(id)}
            >
              <td className='px-6 py-4 border-b border-b-[#FBC707] truncate'>
                {(displayed_title || title).length > 60
                  ? `${(displayed_title || title).slice(0, 60)}...`
                  : displayed_title || title}
              </td>
              <td className='px-6 py-4 border-b border-b-[#FBC707] truncate'>
                {user?.length > 10 ? `${user.slice(0, 10)}...` : user}
              </td>
              <td className='px-6 py-4 flex justify-start'>
                {tags.map((tag, tagIndex) => (
                  <button
                    key={tagIndex}
                    className='bg-[#FBC707] text-black px-3 py-3 rounded-3xl text-xs font-bold mr-3'
                  >
                    {tag}
                  </button>
                ))}
              </td>
              <td className='px-6 py-4 border-b border-b-[#FBC707]'>{timestamp}</td>
              <td className='px-6 py-4 text-right border-b border-b-[#FBC707]'>
                <span
                  className='text-blue-600 underline cursor-pointer hover:text-sm'
                  data-testid={`view-button-${id}`}
                  onClick={(e) => {
                    e.stopPropagation()
                    handleRowClick(id)
                  }}
                >
                  Lihat
                </span>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  )
}

export default AdminTable
