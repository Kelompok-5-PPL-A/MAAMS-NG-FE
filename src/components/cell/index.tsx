import React from 'react'
import { CauseStatus } from '../../lib/enum'
import { CellProps } from '../types/cell'

export const Cell: React.FC<CellProps> = ({
  cellName,
  cause,
  onChange,
  causeStatus,
  disabled,
  placeholder,
  feedback
}) => {
  const getOutlineClass = (status: CauseStatus) => {
    switch (status) {
      case CauseStatus.Incorrect:
        return 'border-red-500'
      case CauseStatus.CorrectNotRoot:
        return 'border-green-500'
      case CauseStatus.CorrectRoot:
        return 'border-purple-500'
      case CauseStatus.Resolved:
        return 'border-gray-200'
      case CauseStatus.Unchecked:
      default:
        return 'border-black'
    }
  }

  const outlineClass = getOutlineClass(causeStatus)

  const renderFeedback = () => {
    let emoji = ''
    let color = ''
    let feedbackText = feedback || ''
    
    switch (causeStatus) {
      case CauseStatus.CorrectRoot:
        emoji = '☑️'
        color = 'purple'
        if (!feedbackText.includes('Akar Masalah')) {
          feedbackText = `Akar Masalah Kolom ${cellName[0]} ditemukan: ${feedbackText}`
        }
        break
      case CauseStatus.CorrectNotRoot:
        emoji = '✅'
        color = 'green'
        break
      case CauseStatus.Incorrect:
        emoji = '❌'
        color = 'red'
        break
      default:
        return null
    }

    return (
      <div className='feedback-text mt-4' style={{ color: color }}>
        {emoji} {feedbackText}
      </div>
    )
  }

  return (
    <div className='flex flex-col items-center justify-center relative'>
      <div className='relative w-fit mt-[-1.00px] font-bold text-black text-2xl leading-10 mb-2 whitespace-nowrap'>
        {cellName}
      </div>
      <textarea
        value={cause}
        onChange={(event) => {
          !disabled && onChange(event.target.value)
        }}
        rows={1}
        maxLength={148}
        className={`w-full h-22 text-xs resize-none flex pt-4 px-4 pb-16 items-center bg-[#ececec] border-solid border-2 ${outlineClass} relative z-[1]`}
        placeholder={placeholder}
        disabled={disabled}
      ></textarea>
      {renderFeedback()}
    </div>
  )
}