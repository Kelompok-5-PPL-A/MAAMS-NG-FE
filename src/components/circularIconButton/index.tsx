import React from 'react'
import { CircularIconButtonProps } from '../types/circularIconButtonProps'

export const CircularIconButton: React.FC<CircularIconButtonProps> = ({ id, icon, onClick, type }) => {
  return (
    <div className='w-[52px] h-[52px] bg-yellow-400 rounded-full flex justify-center items-center'>
      <button data-testid='submit-question' id={id} type={type} color='black' onClick={onClick}>
        {icon}
      </button>
    </div>
  )
}
