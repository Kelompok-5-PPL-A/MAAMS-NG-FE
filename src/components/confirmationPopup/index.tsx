import React, { useState, useEffect } from 'react'
import { ConfirmationPopupProps } from '../types/confirmation'
import Mode from '../../constants/mode'

const Dialog: React.FC<{ isOpen: boolean; onClose: () => void; children: React.ReactNode }> = ({ isOpen, onClose, children }) => {
    useEffect(() => {
      const handleKeyDown = (e: KeyboardEvent) => {
        if (e.key === 'Escape') {
          onClose()
        }
      }
      if (isOpen) {
        document.addEventListener('keydown', handleKeyDown)
      }
      return () => {
        document.removeEventListener('keydown', handleKeyDown)
      }
    }, [isOpen, onClose])
  
    if (!isOpen) return null
  
    return (
      <div
        className='fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center z-50' 
        onClick={onClose} 
        role='dialog' 
        aria-modal='true' 
        tabIndex={0}
        onKeyDown={(e) => e.key === 'Enter' && onClose()}
      >
        <div
          className='bg-white w-[400px] max-w-full p-6 rounded-lg shadow-lg overflow-auto text-center' 
          onClick={(e) => e.stopPropagation()} 
          role='document'
          tabIndex={0}
        >
          {children}
        </div>
      </div>
    )
  }

const Button: React.FC<{ onClick: () => void; variant?: 'outline' | 'gradient'; children: React.ReactNode }> = ({
  onClick,
  variant = 'outline',
  children,
}) => {
  const baseClass = 'py-2 px-12 rounded-xl font-semibold'
  const variantClass = variant === 'gradient' ? 'bg-gradient-to-b from-yellow-400 to-yellow-600 text-l text-white font-bold py-2 px-12 rounded-xl' : 'border border-yellow-400 text-yellow-400'

  return (
    <button onClick={onClick} className={`${baseClass} ${variantClass}`}>
      {children}
    </button>
  )
}

const ConfirmationPopup: React.FC<ConfirmationPopupProps> = ({ mode, onConfirm, onCancel }) => {
  const [isOpen, setIsOpen] = useState(true)

  const handleConfirm = () => {
    onConfirm()
    setIsOpen(false)
  }

  const handleCancel = () => {
    onCancel()
    setIsOpen(false)
  }

  const getMessage = () => {
    return mode === Mode.pengawasan
      ? 'Apakah Anda yakin ingin menampilkan analisis ini kepada Admin?'
      : 'Ubah analisis menjadi pribadi?'
  }

  return (
    <Dialog isOpen={isOpen} onClose={handleCancel}>
      <h2 className='text-xl font-bold mb-4'>{getMessage()}</h2>
      <div className='flex justify-center gap-4'>
        <Button variant='outline' onClick={handleCancel}>
          Batal
        </Button>
        <Button variant='gradient' onClick={handleConfirm}>
          Simpan
        </Button>
      </div>
    </Dialog>
  )
}

export default ConfirmationPopup
