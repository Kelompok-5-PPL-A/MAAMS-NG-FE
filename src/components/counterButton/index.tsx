import { CounterButtonProps } from '../types/counterButton'
import React from 'react'

export const CounterButton: React.FC<CounterButtonProps> = ({ number, onIncrement, onDecrement }) => {
  const textStyle = 'text-2xl font-bold relative text-center whitespace-nowrap z-[1]'
  const decrementBorderBase =
    'border-r-0 flex h-10 py-2 px-8 justify-center items-center shrink-0 flex-nowrap rounded-tl-[10px] rounded-tr-none rounded-br-none rounded-bl-[10px] border-solid border relative'
  const incrementBorderBase =
    'border-l-0 flex h-10 py-2 px-8 justify-center items-center shrink-0 flex-nowrap rounded-tl-none rounded-tr-[10px] rounded-br-[10px] rounded-bl-none border-solid border relative z-[4]'

  let decrementStyle = 'text-[#fbc707] ' + textStyle
  let incrementStyle = 'text-[#fbc707] ' + textStyle
  let decrementBorderStyle = 'border-[#fbc707] ' + decrementBorderBase
  let incrementBorderStyle = 'border-[#fbc707] ' + incrementBorderBase

  if (number === 3) {
    decrementStyle = 'text-[#aeaeae] ' + textStyle
    decrementBorderStyle = 'border-[#aeaeae] ' + decrementBorderBase
  } else if (number === 5) {
    incrementStyle = 'text-[#aeaeae] ' + textStyle
    incrementBorderStyle = 'border-[#aeaeae] ' + incrementBorderBase
  }

  return (
    <div className='items-center my-2'>
      <h2 className='text-center text-sm font-bold text-black relative mx-auto mb-2'>Jumlah Kolom (Max 5)</h2>
      <div className='main-container flex justify-center items-center flex-nowrap relative mx-auto my-0'>
        <button onClick={onDecrement} className={decrementBorderStyle}>
          <span className={decrementStyle}>-</span>
        </button>
        <div className='flex h-10 py-2 px-8 justify-center items-center shrink-0 flex-nowrap bg-[#fbc707] relative z-[2]'>
          {number >= 0 && (
            <span className='shrink-0 basis-auto text-[14px] font-bold text-black relative text-left whitespace-nowrap z-[3]'>
              {number}
            </span>
          )}
        </div>
        <button onClick={onIncrement} className={incrementBorderStyle}>
          <span className={incrementStyle}>+</span>
        </button>
      </div>
    </div>
  )
}
