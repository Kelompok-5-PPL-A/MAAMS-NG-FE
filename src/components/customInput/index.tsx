import { CustomInputProps } from '../types/customInput'
import React from 'react'
import { Input, Icon, InputGroup } from '@chakra-ui/react'
import { WarningTwoIcon } from '@chakra-ui/icons'

export const CustomInput: React.FC<CustomInputProps> = ({
  placeholder,
  label,
  labelClassName,
  inputClassName,
  errorClassName,
  spacerClassName,
  onChange,
  onKeyDown,
  value,
  children,
  error,
  isDisabled
}) => {
  const getIcon = () => {
    if (error) {
      return <Icon as={WarningTwoIcon} color='red.500' />
    } else {
      return null
    }
  }

  return (
    <div className={`${spacerClassName ?? 'space-y-4'} w-full`}>
      {!!label && <label className={`${labelClassName}`}>{label}</label>}
      <InputGroup className='border-zinc-500'>
        <Input
          placeholder={placeholder}
          onChange={onChange}
          onKeyDown={onKeyDown}
          value={value}
          className={`${inputClassName} disabled:text-black`}
          disabled={isDisabled}
          variant={isDisabled ? 'filled' : 'outline'}
        />
        {children}
      </InputGroup>
      <div className='flex items-center gap-4'>
        {getIcon()}
        {!!error && <label className={`${errorClassName}`}>{error}</label>}
      </div>
    </div>
  )
}