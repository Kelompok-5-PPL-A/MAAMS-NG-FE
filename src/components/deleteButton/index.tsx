import React, { useRef, useState } from 'react'
import { useEffect } from 'react'
import { useOnClickOutside } from 'usehooks-ts'
import { HiDotsVertical, HiTrash } from 'react-icons/hi'
import { DeleteButtonProps } from '../types/deleteButtonProps'
import { Modal, ModalOverlay, ModalContent, ModalFooter, ModalBody, ModalCloseButton } from '@chakra-ui/react'
import axiosInstance from '../../services/axiosInstance'
import toast from 'react-hot-toast'
import router from 'next/router'

export const DeleteButton = ({ idQuestion, pathname }: DeleteButtonProps) => {
  const [isOpen, setIsOpen] = useState<boolean>(false)
  const ref = useRef<HTMLDivElement | null>(null)
  const handleModalDeleteOpen = () => setIsModalDeleteOpen(!isModalDeleteOpen)
  const [isModalDeleteOpen, setIsModalDeleteOpen] = useState<boolean>(false)

  useEffect(() => {
    setIsOpen(false)
  }, [])

  useOnClickOutside(ref as React.RefObject<HTMLElement>, () => {
    setIsOpen(false)
  })

  const handleDelete = async () => {
    try {
      await axiosInstance.delete(`/api/v1/validator/hapus/${idQuestion}/`)
      toast.success('Berhasil menghapus analisis')
      if (pathname === '/history') {
        router.reload()
      } else {
        router.push('/history')
      }
    } catch (error: any) {
      if (error.response) {
        toast.error(error.response.data.detail)
      } else {
        toast.error('Gagal menghapus analisis')
      }
    }
  }

  return (
    <>
      <div className='relative' ref={ref}>
        <div
          onClick={() => setIsOpen(!isOpen)}
          data-testid='toggle-open-button'
          className={`cursor-pointer py-2 text-md font-bold flex items-center justify-between`}
        >
          <button className='w-fit h-fit' type='button'>
            <HiDotsVertical className='w-4 h-4' />
          </button>
          {isOpen && (
            <div
              className='absolute w-[200px] rounded-xl top-8 right-0 mt-2 w-56 bg-slate-100 shadow-lg text-red'
              role='menu'
              aria-orientation='vertical'
              aria-labelledby='options-menu'
            >
              <div className='py-1'>
                <button
                  onClick={handleModalDeleteOpen}
                  className='block px-4 py-4 text-sm cursor-pointer justify-center items-center w-full'
                  data-testid='delete-button'
                >
                  <div className='text-red-600 flex flex-row gap-2 justify-center items-center'>
                    <HiTrash className='w-6 h-6' />
                    <p>Hapus analisis</p>
                  </div>
                </button>
              </div>
            </div>
          )}
        </div>
      </div>

      <Modal isOpen={isModalDeleteOpen} onClose={handleModalDeleteOpen}>
        <ModalOverlay />
        <ModalContent className='py-8'>
          <ModalCloseButton />
          <ModalBody className='items-center mt-8 mx-4 text-center text-xl font-bold'>
            Apakah Anda yakin ingin menghapus analisis ini?
          </ModalBody>
          <ModalFooter>
            <div className='w-full flex flex-row gap-4'>
              <button
                className='w-full px-6 py-2 border-2 border-yellow-400 rounded-2xl justify-center items-center text-black text-lg'
                onClick={handleModalDeleteOpen}
              >
                Batal
              </button>
              <button
                className='w-full px-6 py-2 bg-gradient-to-t from-red-500 to-red-900 text-white  rounded-2xl justify-center items-center gap-2 inline-flex'
                onClick={handleDelete}
              >
                Hapus
              </button>
            </div>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </>
  )
}
