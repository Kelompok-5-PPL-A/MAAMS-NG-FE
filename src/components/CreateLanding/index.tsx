import { CustomInput } from '../customInput'
import Link from 'next/link'
import React, { useEffect, useState } from 'react'
import { useRouter } from 'next/router'
import toast from 'react-hot-toast'

const CreateLanding = () => {
  const [isLoggedIn, setIsLoggedIn] = useState(false)
  const [question, setQuestion] = useState<string>('')
  const router = useRouter()

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()

    if (!question) {
      toast.error('Pertanyaan harus diisi')
      return
    }

    router.push({
      pathname: '/validator',
      query: { question: question }
    })
  }

  useEffect(() => {
    const refresh_token = localStorage.getItem('isLoggedIn')
    setIsLoggedIn(refresh_token === 'true')
  })

  return (
    <div className='flex flex-col lg:flex-row justify-between items-center gap-12 lg:mx-55'>
      <div className=''>
        <img src='/icons/landing-icon.svg' alt='landing' className='' />
      </div>
      <div className=''>
          <div className='flex flex-col gap-8 mx-12'>
            <h1 className='text-3xl font-bold text-center'>Apa masalah yang ingin dianalisis hari ini?</h1>
            <div className='flex flex-row gap-2'>
              <form className='flex flex-row w-full gap-x-4 items-center justify-center' onSubmit={handleSubmit}>
                <CustomInput
                  inputClassName='p-6 text-sm w-[400px] border border-black-300 rounded-lg shadow-md focus:shadow-lg transition-shadow duration-200'
                  placeholder='ingin menganalisis apa hari ini ...'
                  spacerClassName='space-y-0'
                  isDisabled={false}
                  onChange={(e) => setQuestion(e.target.value)}
                  value={question}
                />
                <button className='' title='submit_button'>
                  <img src='/icons/send-icon.svg' alt='search' className='bg-yellow-400 p-3 w-20 h-15 rounded-full' />
                </button>
              </form>
            </div>
          </div>
      </div>
    </div>
  )
}

export default CreateLanding