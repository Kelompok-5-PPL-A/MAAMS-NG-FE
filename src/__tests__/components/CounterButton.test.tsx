import '@testing-library/jest-dom'
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { CounterButton } from '../../components/counterButton'

describe('CounterButton Component', () => {
  test('renders with the given number', () => {
    const number = 3
    const onIncrement = jest.fn()
    const onDecrement = jest.fn()

    const { getByText } = render(<CounterButton number={number} onIncrement={onIncrement} onDecrement={onDecrement} />)
    expect(getByText(number.toString())).toBeInTheDocument()
  })

  test('calls onIncrement when the "+" button is clicked', () => {
    const onIncrement = jest.fn()
    const onDecrement = jest.fn()

    const { getByText } = render(<CounterButton number={3} onIncrement={onIncrement} onDecrement={onDecrement} />)

    fireEvent.click(getByText('+'))
    expect(onIncrement).toHaveBeenCalled()
  })

  test('calls onDecrement when the "-" button is clicked', () => {
    const onIncrement = jest.fn()
    const onDecrement = jest.fn()

    const { getByText } = render(<CounterButton number={3} onIncrement={onIncrement} onDecrement={onDecrement} />)

    fireEvent.click(getByText('-'))
    expect(onDecrement).toHaveBeenCalled()
  })

  test('does not render with a negative number', () => {
    const onIncrement = jest.fn()
    const onDecrement = jest.fn()

    const { queryByText } = render(<CounterButton number={-1} onIncrement={onIncrement} onDecrement={onDecrement} />)
    expect(queryByText((content, element) => content === '-1' && !element?.className.includes('text-black'))).toBeNull()
  })

  test('renders with default styles and border color when number is 4', () => {
    const { getByText } = render(<CounterButton number={4} onIncrement={() => {}} onDecrement={() => {}} />)

    const incrementButton = getByText('+')
    const decrementButton = getByText('-')

    expect(incrementButton).toHaveClass('text-[#fbc707]')
    expect(decrementButton).toHaveClass('text-[#fbc707]')
    expect(incrementButton.parentElement).toHaveClass('border-[#fbc707]')
    expect(decrementButton.parentElement).toHaveClass('border-[#fbc707]')
  })

  test('renders with disabled styles and border color when number is 3', () => {
    const { getByText } = render(<CounterButton number={3} onIncrement={() => {}} onDecrement={() => {}} />)

    const incrementButton = getByText('+')
    const decrementButton = getByText('-')

    expect(incrementButton).toHaveClass('text-[#fbc707]')
    expect(decrementButton).toHaveClass('text-[#aeaeae]')
    expect(incrementButton.parentElement).toHaveClass('border-[#fbc707]')
    expect(decrementButton.parentElement).toHaveClass('border-[#aeaeae]')
  })

  test('renders with disabled styles and border color when number is 5', () => {
    const { getByText } = render(<CounterButton number={5} onIncrement={() => {}} onDecrement={() => {}} />)

    const incrementButton = getByText('+')
    const decrementButton = getByText('-')

    expect(incrementButton).toHaveClass('text-[#aeaeae]')
    expect(decrementButton).toHaveClass('text-[#fbc707]')
    expect(incrementButton.parentElement).toHaveClass('border-[#aeaeae]')
    expect(decrementButton.parentElement).toHaveClass('border-[#fbc707]')
  })

  test('does not have incorrect styles when number is 4', () => {
    const { getByText } = render(<CounterButton number={4} onIncrement={() => {}} onDecrement={() => {}} />)

    const incrementButton = getByText('+')
    const decrementButton = getByText('-')

    expect(incrementButton).not.toHaveClass('text-red-500')
    expect(decrementButton).not.toHaveClass('text-red-500')
  })

  test('renders correctly with number 0', () => {
    const { getByText } = render(<CounterButton number={0} onIncrement={() => {}} onDecrement={() => {}} />)
    expect(getByText('0')).toBeInTheDocument()
  })

  test('matches snapshot', () => {
    const { asFragment } = render(<CounterButton number={3} onIncrement={() => {}} onDecrement={() => {}} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
