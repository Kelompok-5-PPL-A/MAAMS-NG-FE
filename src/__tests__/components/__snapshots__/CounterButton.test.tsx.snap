// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="items-center my-2"
  >
    <h2
      class="text-center text-sm font-bold text-black relative mx-auto mb-2"
    >
      Jumlah Kolom (Max 5)
    </h2>
    <div
      class="main-container flex justify-center items-center flex-nowrap relative mx-auto my-0"
    >
      <button
        class="border-[#aeaeae] border-r-0 flex h-10 py-2 px-8 justify-center items-center shrink-0 flex-nowrap rounded-tl-[10px] rounded-tr-none rounded-br-none rounded-bl-[10px] border-solid border relative"
      >
        <span
          class="text-[#aeaeae] text-2xl font-bold relative text-center whitespace-nowrap z-[1]"
        >
          -
        </span>
      </button>
      <div
        class="flex h-10 py-2 px-8 justify-center items-center shrink-0 flex-nowrap bg-[#fbc707] relative z-[2]"
      >
        <span
          class="shrink-0 basis-auto text-[14px] font-bold text-black relative text-left whitespace-nowrap z-[3]"
        >
          3
        </span>
      </div>
      <button
        class="border-[#fbc707] border-l-0 flex h-10 py-2 px-8 justify-center items-center shrink-0 flex-nowrap rounded-tl-none rounded-tr-[10px] rounded-br-[10px] rounded-bl-none border-solid border relative z-[4]"
      >
        <span
          class="text-[#fbc707] text-2xl font-bold relative text-center whitespace-nowrap z-[1]"
        >
          +
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
